HeatEquationSolver/heat_equation.py

```diff
@@ -20,28 +30,82 @@ def initialise_grid(length, nodes, time, k):
     return dx, dt, time_step, x, t
 
 
-def setting_conditions(t, time_step, nodes):
-    # initial and boundary conditions
-    u_initial = 25
-    u_left = 20 * np.sin(np.pi * t) + 25
-    u_right = t + 25
+def left_temp_func(temp, t):
+    """
+    Generates the temperature at the left boundary of the rod
+
+    :param temp: initial temperature of the rod at t=0
+    :param t: array consisting of time steps from t=0 to t= total time
+    :return: the temperature at the left boundary
+    """
+    return 20 * np.sin(np.pi * t) + temp
+
+
+def right_temp_func(temp, t):
+    """
+    Generates the temperature at the right boundary of the rod
+
+    :param temp: initial temperature of the rod at t=0
+    :param t: array consisting of time steps from t=0 to t= total time
+    :return: the temperature at the right boundary
+    """
+    return t + temp
+
+
+def setting_conditions(t, time_step, nodes, temp):
+    """
+    Initialises the temperature distribution and boundary conditions for the heat equation
+
+    :param t: array consisting of time steps from t=0 to t=total time
+    :param time_step: no of time steps for the simulation
+    :param nodes: no of spatial nodes
+    :param temp: initial temperature of the rod at t=0
+    :return:    - u: 2d-matrix initialised with the initial temperature of the rod
+                - u_left: array denoting the temperature at the left boundary over time
+                - u_right: array denoting the temperature at the right boundary over time
+    """
+    u_initial = temp
+    u_left = left_temp_func(temp, t)
+    u_right = right_temp_func(temp, t)
     u = np.zeros((time_step, nodes))
     u[0, :] = u_initial
 
     return u, u_left, u_right
 
 
-# initializing 2d-matrix to store temperature
 def solve_heat_equation(u, u_left, u_right, time_step, nodes, dt, k, dx):
+    """
+    Solves the heat equation using the explicit scheme
+
+    :param u: 2d-matrix initialised with the initial temperature of the rod
```

Comment:
If `u` has been initialised already, why do you need to pass `u_left` and `u_right` too?

Reply:
the 'u' matrix im passing into the function only has the initial temp initialised, not the left + right boundary temperatures

Reply:
Why not left and right as well? Here's a suggestion: create a function of the form:

```python
grid_params = {"dx": ..., "dt": ...} # you can make this a class later (you can choose a non-uniform grid etc)
solver_params = {"type": "cn"}
left = lambda t: some_function(t)
right = lambda t: some_function(t)
init = lambda x: some_function(x)
sol = solve_heat_equation(kappa, grid_params, solver_params, left, right, init)
# sol is a dictionary/class object with the solution
# if it is a class object, you can do
sol.plot()
# you can do things like
mean_temp = sol.mean_temperature()
# and you can also compare two solutions
error = sol1.diff(sol2)
```

That said, your don't need to do this now. We can merge it now and update it later. Your call.

Reply:
will look into it, thank you !

Reply:
@chemardes what did you decide?

Reply:
i think i'll use your implementation - makes things more structured!